hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieParquetInputFormat.java

```diff
@@ -52,12 +59,36 @@ public class HoodieParquetInputFormat extends HoodieParquetInputFormatBase {
 
   private static final Logger LOG = LoggerFactory.getLogger(HoodieParquetInputFormat.class);
 
+  private boolean supportAvroRead = true;
+
   public HoodieParquetInputFormat() {
     super(new HoodieCopyOnWriteTableInputFormat());
+    initAvroInputFormat();
   }
 
   protected HoodieParquetInputFormat(HoodieCopyOnWriteTableInputFormat delegate) {
     super(delegate);
+    initAvroInputFormat();
+  }
+
+  /**
+   * Spark2 use `parquet.hadoopParquetInputFormat` in `com.twitter:parquet-hadoop-bundle`.
+   * So that we need to distinguish the constructions of classes with
+   * `parquet.hadoopParquetInputFormat` or `org.apache.parquet.hadoop.ParquetInputFormat`.
+   * If we use `org.apache.parquet:parquet-hadoop`, we can use `HudiAvroParquetInputFormat`
+   * in Hive or Spark3 to get timestamp with correct type.
+   */
+  private void initAvroInputFormat() {
+    try {
+      Constructor[] constructors = ParquetRecordReaderWrapper.class.getConstructors();
+      if (Arrays.stream(constructors)
+          .anyMatch(c -> c.getParameterCount() > 0 && c.getParameterTypes()[0]
+              .getName().equals(ParquetInputFormat.class.getName()))) {
+        supportAvroRead = true;
```

Comment:
when will `supportAvroRead` be false?

Reply:
@Zouxxyy You can read the comments. This is to address compatibility issues with spark

Reply:
> @Zouxxyy You can read the comments. This is to address compatibility issues with spark

yeal, but it always be true now

Reply:
In my impression, hive2 and spark3 have the same processing method and constructor, but there is a difference in hive3

Reply:
We should revisit this method, the name `ParquetInputFormat.class.getName()` in hive2 is the same as hive3.

Reply:
This is for compatibility with spark，do you mean hive on spark? @cdmikechen